### DISCUSSION TITLE
Question about `useWatch`

### DISCUSSION THREAD

**Discussion by Dragomir-Ivanov** [react-hook-form/react-hook-form#8140] [Q&A] [answered]
Hi guys,
Looking at the documentation I don't quite understand what, following is supposed to be used, and what is its purpose. Any practical examples, or even Docs update appreciated. Thank you!
```
useWatch({
....
disabled: XXX
})

**Comment by Moshyfawn** [comment, marked as answer]
It's basically a conditional subscription. The same, as, say this _(pseudo-code)_
```js
if (isChecked) {
  fieldValue.subscribe()
} else {
  fieldValue.unsubscribe()
}
```

Just a convenience prop to toggle the field values subscription _(an optimization measure)_

If you have some ideas on how to improve the docs to allow better understanding of the feature, we're open to suggestions!

**Reply by roksui** [reply]
What is the `fieldValue` exactly?

**Reply by BrendanC23** [reply]
`fieldValue` here is just psuedocode for whichever field name you pass into `useWatch()`. If you call

```
function Component() {
    const firstName = useWatch({
        control,
        name: "firstName",
        disabled: true
    });

    return (<></>);
}
```

then `<Component />` will not subscribe to changes to `firstName` (until `disabled` is changed to back to false). If `firstName` changes, `<Component />` will not re-render. If a different component uses `useWatch({ name: "firstName", disabled: false })`, that component will still re-render.
